testutil-server/src/main/java/io/spine/testing/server/blackbox/BlackBoxContext.java

```diff
@@ -492,58 +505,130 @@ public void close() {
      */
-    public <I, E extends Entity<I, ? extends EntityState>>
+    @Deprecated
+    public final <I, E extends Entity<I, ? extends EntityState>>
     EntitySubject assertEntity(Class<E> entityClass, I id) {
-        @Nullable Entity<I, ?> found = findEntity(entityClass, id);
+        return assertEntity(id, entityClass);
+    }
+
+    /**
+     * Obtains a Subject for an entity of the passed class with the given ID.
+     */
+    public final <I, E extends Entity<I, ? extends EntityState>>
+    EntitySubject assertEntity(I id, Class<E> entityClass) {
+        @Nullable Entity<I, ?> found = findEntity(id, entityClass);
         return EntitySubject.assertEntity(found);
     }
 
     private <I> @Nullable Entity<I, ?>
-    findEntity(Class<? extends Entity<I, ?>> entityClass, I id) {
+    findEntity(I id, Class<? extends Entity<I, ?>> entityClass) {
         Class<? extends EntityState> stateClass = stateClassOf(entityClass);
-        return findByState(stateClass, id);
+        return findByState(id, stateClass);
     }
 
     /**
      * Obtains a Subject for an entity which has the state of the passed class with the given ID.
+     *
+     * @deprecated please use {@link #assertEntityWithState(Object, Class)}
      */
-    public <I, S extends EntityState> EntitySubject
+    @Deprecated
+    public final <I, S extends EntityState> EntitySubject
     assertEntityWithState(Class<S> stateClass, I id) {
-        @Nullable Entity<I, S> found = findByState(stateClass, id);
+        return assertEntityWithState(id, stateClass);
+    }
+
+    /**
+     * Obtains a Subject for an entity which has the state of the passed class with the given ID.
+     */
+    public final <I, S extends EntityState>
+    EntitySubject assertEntityWithState(I id, Class<S> stateClass) {
+        @Nullable Entity<I, S> found = findByState(id, stateClass);
         return EntitySubject.assertEntity(found);
     }
 
     private <I, S extends EntityState> @Nullable Entity<I, S>
-    findByState(Class<S> stateClass, I id) {
+    findByState(I id, Class<S> stateClass) {
         @SuppressWarnings("unchecked")
         Repository<I, ? extends Entity<I, S>> repo =
                 (Repository<I, ? extends Entity<I, S>>) repositoryOf(stateClass);
         return readOperation(() -> (Entity<I, S>) repo.find(id).orElse(null));
     }
 
     @VisibleForTesting
-    Repository<?, ?> repositoryOf(Class<? extends EntityState> stateClass) {
+    final Repository<?, ?> repositoryOf(Class<? extends EntityState> stateClass) {
         Repository<?, ?> repository =
                 context.internalAccess()
                        .getRepository(stateClass);
         return repository;
     }
 
+    /**
+     * Asserts that the is an entity with the passed ID and the passed type of state.
+     *
+     * @return assertion that compares only expected fields
+     */
+    public final <I, S extends EntityState> ProtoFluentAssertion
+    assertState(I id, Class<S> stateClass) {
```

Comment:
I suggest we get rid of unnecessary type parameters. In this case, since neither `<I>` nor `<S>` is used in the return type, they both are unnecessary.

Also, there used to be a style check for that. Is it off now?

Reply:
Indeed, they are not used. But they are consistent with the rest of assertion methods. Here having `<I>` as a generic parameter we hint on the fact that an object which can be passed to it can be of one of the identifier types, not any `Object`.